transport/http2_server.go

```diff
@@ -1022,9 +992,39 @@ func (t *http2Server) itemHandler(i item) error {
 			i.f()
 		}
 	case *headerFrame:
-		err = t.framer.fr.WriteHeaders(i.p)
-	case *continuationFrame:
-		err = t.framer.fr.WriteContinuation(i.streamID, i.endHeaders, i.headerBlockFragment)
+		t.hBuf.Reset()
```

Comment:
Can this logic be shared with the client's?

transport.go or control.go: `headerFrameHandler(hEnc, hBuf, framer)`

Reply:
Actually, most of these cases in item handler can be shared between the server and the client along with most of Write method's code and I'm guessing there are other such opportunities too scattered throughout the transport layer. I kept it this way to be consistent with the rest of the code and with hopes that one day we'll refactor the transport to take care of all this duplication.

Reply:
SGTM, let's try not to put it off for too long.  Maybe add a TODO in this PR...